src/main/java/com/afornalik/ox/board/Board.java

```diff
@@ -22,15 +22,20 @@ public int getBound() {
         return bound;
     }
 
+    public void insertBoardField(int indexOfField, FieldStatus fieldStatusX) throws OutOfBoardException {
+        if (indexOfField < minValue || indexOfField > maxValue.intValue()) {
+            throw new OutOfBoardException("Out of board - min value is : " + minValue + ", and max value is : " + maxValue.toString());
+        }
+        BoardField boardField = new BoardField(fieldStatusX);
+        boardFields.put(indexOfField, boardField);
+    }
 
-    BoardField receiveBoardField(int i) throws OutOfBoardException {
-        if (i < minValue || i > maxValue.intValue()) {
-            throw new OutOfBoardException("Out of board - min value is : " +minValue+", and max value is : "+maxValue.toString());
-        } else if (boardFields.get(i) == null) {
+    public BoardField receiveBoardField(int indexOfField) throws OutOfBoardException {
+        if (indexOfField < minValue || indexOfField > maxValue.intValue()) {
+            throw new OutOfBoardException("Out of board - min value is : " + minValue + ", and max value is : " + maxValue.toString());
+        } else if (boardFields.get(indexOfField) == null) {
             return null;
         }
-        return boardFields.get(1);
+        return boardFields.get(indexOfField);
```

Comment:
I see some code duplication, extract it to some private helper method. Also consider not returning a null value (to avoid NPE)

Reply:
Also, board is public and have no documentation

Reply:
I added documentation, and also i extracted private method to avoid duplication, method will no longer return null reference.

Reply:
Great job!